### PR TITLE
Add default option on head requests

### LINKED ISSUE #345: Head method returns the request content
When trying to use the HEAD method I'm getting the request content while the correct is to only get the headers:

```
[9] pry(main)> HTTParty.head('http://google.com/any-page')
=> "<!DOCTYPE html>\n<html lang=en>\n  <meta charset=utf-8>\n  <meta name=viewport content=\"initial-scale=1, minimum-scale=1, width=device-width\">\n  <title>Error 404 (Not Found)!!1</title>\n  <style>\n    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/errors/logo_sm_2.png) no-repeat}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/errors/logo_sm_2_hr.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/errors/logo_sm_2_hr.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/errors/logo_sm_2_hr.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:55px;width:150px}\n  </style>\n  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>\n  <p><b>404.</b> <ins>That’s an error.</ins>\n  <p>The requested URL <code>/any-page</code> was not found on this server.  <ins>That’s all we know.</ins>\n"
````

After testing in a local server, it looks like httparty sends two requests with different methods: HEAD and GET. Sample Apache's logs:

```
192.168.0.20 (-) - - [28/Nov/2014:00:43:42 +0100] "HEAD / HTTP/1.1" 302 - "-" "Ruby"
192.168.0.20 (-) - - [28/Nov/2014:00:43:43 +0100] "GET / HTTP/1.1" 302 - "-" "Ruby"
```

Any ideas of what might cause this problem?

### PR DESCRIPTION
**Pull request by supremebeing7**
This attempts to resolve #345. I'm pretty confident the code works, but I couldn't get the spec figured out. I left my attempt in there, with the expectation line commented out. 

Suggestions welcome/encouraged.

### CODE REVIEW

**Comment by jnunemaker** [comment]
I'm not comfortable merging without a passing spec and I don't have time to look into this right now. If someone else could help, that would be great.

**Comment by supremebeing7** [comment]
Okay, I had some more time and was able to refactor this a bit to make it more easily testable, then add specs. 

I went with the "principle of least surprise" approach [outlined by xunker](https://github.com/jnunemaker/httparty/issues/345#issuecomment-97261818), ensuring that head requests maintain the HEAD method across all redirects, unless the user specifies `maintain_method_across_redirects: false` in the options.

File: lib/httparty.rb [hunk 1]
```diff
@@ -529,6 +532,12 @@ def options(path, options = {}, &block)
 
     private
 
+    def ensure_method_maintained_across_redirects(options)
+      unless options.has_key? :maintain_method_across_redirects
```
Comment (jnunemaker) [inline lib/httparty.rb]: This means it will do a GET if it is set to false, is that right?

Comment (supremebeing7) [inline lib/httparty.rb]: Exactly. I wanted to make sure there was still a way to override the new behavior, in case someone wanted to do that.

**Comment by xunker** [comment]
@supremebeing7 Thanks for taking care of this, I'm glad my research was helpful to you. :+1:

**Comment by supremebeing7** [comment]
@xunker Very helpful :tada:

**Pull request merged**